packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java

```diff
@@ -888,8 +904,30 @@ private void destroyMapViewIfNecessary() {
     if (mapView == null) {
       return;
     }
-    mapView.onDestroy();
+
+    final MapView mapReference = mapView; // keep a reference to the mapView for the callback
     mapView = null;
+
+    // This fixes an issue where the mapView is still being used by the render thread after disposal.
```

Comment:
Nit: replace "fixes" with "attempts to work around"; this isn't actually a fix, since it won't reliably work depending on internal details. This should also link to the issue for context.